### ISSUE TITLE
Error in Naive Bayes with non uniform priors

### ISSUE THREAD

**Issue by devmag** [kbenoit/quanteda#546]
Just to illustrate:
```
trainingset <- as.dfm(matrix(c(1, 2, 0, 0, 0, 0,
                        0, 2, 0, 0, 1, 0,
                        0, 1, 0, 1, 0, 0,
                        0, 1, 1, 0, 0, 1,
                        0, 3, 1, 0, 0, 1), 
                      ncol=6, nrow=5, byrow=TRUE,
                      dimnames = list(docs = paste("d", 1:5, sep = ""),
                                      features = c("Beijing", "Chinese",  "Japan", "Macao", 
                                                   "Shanghai", "Tokyo"))))
trainingclass <- factor(c("Y", "Y", "Y", "N", NA), ordered = TRUE)
## replicate IIR p261 prediction for test set (document 5)
nb.p261 <- textmodel_NB(trainingset, trainingclass)

##uniform priors
nb.p261$Pc

# 
nb2<-textmodel_NB(trainingset, trainingclass, prior = "docfreq")

nb2$Pc

#reports 
#  Y    N 
#0.25 0.75 
```
when in fact it should be N 25 % and Y 75%.

I think that error occurs in step where names are assigned in textmodel_NB function.

**Comment by kbenoit** [comment]
Thanks @devmag - this was found and fixed in issue #476. Corrected version currently under review at CRAN and should be published shortly, but the GitHub version of course has fixed this already (ca927af).

**Comment by devmag** [comment]
Thanks Ken!
